### PR TITLE
prov/rxd: Cleanup of provider code WIP

### PR DESCRIPTION
**Pull request by shefty**
This is a significant restructuring and cleanup of the RXD code, and is a work in progress.  Posting so that other developers can build off of these changes.  The primary focus of these changes is to change the progress model from auto to manual, so that it can make use of the common util code.  (The current model does not work at all in a virtual box environment due to thread starvation).

The changes focus solely on getting the message data transfers to work over the underlying provider.  The existing tagged, rma, and atomic code is retained, but no effort has been made to clean up that code.  The code compiles, but no testing has been done.  More cleanup is needed.

The rxd provider was previously disabled.  This leaves it disabled.  This patch should be viewed more in line as replacing the existing provider code, than attempting to evolve it.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>

### CODE REVIEW

**Comment by shefty** [comment]
@gladkovdmitry17 @hoopoepg

File: prov/rxd/src/rxd_ep.c [hunk 1]
```diff
@@ -259,16 +256,13 @@ struct rxd_peer *rxd_ep_getpeer_info(struct rxd_ep *ep, fi_addr_t addr)
 	return &ep->peer_info[addr];
 }
 
-void rxd_ep_lock_if_required(struct rxd_ep *ep)
-{
-	/* todo: do locking based on threading model */
-	fastlock_acquire(&ep->lock);
```
Comment (a-ilango) [inline prov/rxd/src/rxd_ep.c]: Is this not needed anymore?

File: prov/rxd/src/rxd_ep.c [hunk 2]
```diff
@@ -299,19 +293,20 @@ static void rxd_init_op_hdr(struct ofi_op_hdr *op, uint64_t data,
...
+	uint32_t flags = 0;
+
+	if (fi_flags & FI_REMOTE_CQ_DATA)
+		flags = OFI_REMOTE_CQ_DATA;
+	if (fi_flags & FI_TRANSMIT_COMPLETE)
+		flags |= OFI_TRANSMIT_COMPLETE;
+	if (fi_flags & FI_DELIVERY_COMPLETE)
+		flags |= OFI_DELIVERY_COMPLETE;
+	return flags;
 }
```
Comment (a-ilango) [inline prov/rxd/src/rxd_ep.c]: This function can be moved to common code.

**Comment by shefty** [comment]
> Is this not needed anymore?

The code acquires the lock directly.  The optional acquiring of locks wasn't supported and complicates the code.

**Comment by shefty** [comment]
Code was merged via another PR

**Pull request closed without merging**